integration-tests/jpa-mysql test fails against docker instance

integration-tests/jpa-mysql test fails against docker instance - `mvn clean install -Dtest-mysql -Ddocker -pl integration-tests/jpa-mysql`

From the stacktrace the most important message seems `Caused by: java.io.EOFException: Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.`

Tests are passing when running against already running database - `mvn clean install -Dtest-mysql -pl integration-tests/jpa-mysql`

Seems that `-Ddocker` way doesn't wait long enough or there is problem with exposed ports.
Running on macOS 10.15.2, OpenJDK 64-Bit GraalVM CE 19.3.0.

--- a/integration-tests/jpa-mysql/pom.xml
+++ b/integration-tests/jpa-mysql/pom.xml
@@ -195,14 +195,12 @@
                                         <!-- Speed things up a bit by not actually flushing writes to disk -->
                                         <tmpfs>/var/lib/mysql</tmpfs>
                                         <wait>
-                                            <!-- good docs found at: http://dmp.fabric8.io/#build-healthcheck -->
-                                            <tcp>
-                                                <mode>mapped</mode>
-                                                <ports>
-                                                    <port>3306</port>
-                                                </ports>
-                                            </tcp>
-                                            <time>40000</time>
+                                            <!-- good docs found at: http://dmp.fabric8.io/#start-wait -->
+                                            <time>20000</time>
+                                            <!-- wait until MySQL is actually up by checking if mysqladmin can ping the server with specified username/password -->
+                                            <exec>
+                                                <postStart>mysqladmin ping -h localhost -u hibernate_orm_test -phibernate_orm_test</postStart>
+                                            </exec>
                                         </wait>
                                     </run>
                                 </image>

